crates/parcel_napi_helpers/src/transferable.rs

```diff
@@ -0,0 +1,81 @@
+use std::any::Any;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::sync::atomic::AtomicI32;
+use std::sync::atomic::Ordering;
+use std::sync::Mutex;
+
+use napi::bindgen_prelude::FromNapiValue;
+use napi::bindgen_prelude::ToNapiValue;
+use napi::Env;
+use napi::JsNumber;
+use napi::NapiRaw;
+use once_cell::sync::Lazy;
+
+static COUNTER: AtomicI32 = AtomicI32::new(0);
+static VALUES: Lazy<Mutex<HashMap<i32, Box<dyn Any + Send + Sync>>>> =
+  Lazy::new(|| Default::default());
+
+/// Creates an external reference to a Rust value and
+/// makes it transferable across Nodejs workers
+///
+/// This is to get around the limitations of what can be transferred
+/// between workers in Nodejs
+///
+/// https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Transferable_objects
+pub struct JsTransferable<T> {
```

Comment:
Can you add a unit test or demonstrate some integration with an existing structure?

Reply:
I'm currently using`JsTransferable` to transfer a `channel::Sender` from the main thread into the nodejs worker which it uses to notify the main thread that it's ready to be used